src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs

```diff
@@ -120,6 +131,24 @@ protected virtual ControllerModel CreateControllerModel([NotNull] TypeInfo typeI
             return controllerModel;
         }
 
+        /// <summary>
+        /// Creates a <see cref="PropertyModel"/> for the given <see cref="PropertyInfo"/>.
+        /// </summary>
+        /// <param name="propertyInfo">The <see cref="PropertyInfo"/>.</param>
+        /// <returns>A <see cref="PropertyModel"/> for the given <see cref="PropertyInfo"/>.</returns>
+        protected virtual PropertyModel CreatePropertyModel([NotNull] PropertyInfo propertyInfo)
+        {
+            // CoreCLR returns IEnumerable<Attribute> from GetCustomAttributes - the OfType<object>
+            // is needed to so that the result of ToArray() is object
+            var attributes = propertyInfo.GetCustomAttributes(inherit: true).OfType<object>().ToArray();
+            var propertyModel = new PropertyModel(propertyInfo, attributes);
+
+            propertyModel.BinderMetadata = attributes.OfType<IBinderMetadata>().FirstOrDefault();
+            propertyModel.PropertyName = propertyInfo.Name;
```

Comment:
? did not understand, the model binding code does not look at PropertyModel right? they only look at model metadata.